ee/ucp/admin/configure/deploy-route-reflectors.md

```diff
@@ -0,0 +1,261 @@
+
+```
+kubectl label nodes <node-name> \
+  com.docker.ucp.kubernetes.calico/route-reflector=true
+```
+
+## Deploy the Route Reflectors
+
+Create a `calico-rr.yaml` file with the following content:
+
+```
+kind: DaemonSet
+apiVersion: extensions/v1beta1
+metadata:
+  name: calico-rr
+  namespace: kube-system
+  labels:
+    app: calico-rr
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      k8s-app: calico-rr
+  template:
+    metadata:
+      labels:
+        k8s-app: calico-rr
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      tolerations:
+        - key: com.docker.ucp.kubernetes.calico/route-reflector
+          value: "true"
+          effect: NoSchedule
+      hostNetwork: true
+      containers:
+        - name: calico-rr
+          image: calico/routereflector:v0.6.1
+          env:
+            - name: ETCD_ENDPOINTS
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: etcd_endpoints
+            - name: ETCD_CA_CERT_FILE
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: etcd_ca
+            # Location of the client key for etcd.
+            - name: ETCD_KEY_FILE
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: etcd_key # Location of the client certificate for etcd.
+            - name: ETCD_CERT_FILE
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: etcd_cert
+            - name: IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+          volumeMounts:
+            - mountPath: /calico-secrets
+              name: etcd-certs
+          securityContext:
+            privileged: true
+      nodeSelector:
+        com.docker.ucp.kubernetes.calico/route-reflector: "true"
+      volumes:
+      # Mount in the etcd TLS secrets.
+        - name: etcd-certs
+          secret:
+            secretName: calico-etcd-secrets
+  ```
+
+Then, deploy the DaemonSet using:
+
+```
+kubectl create -f calico-rr.yaml
+```
+
+## Configure calicoctl
+
+To reconfigure Calico to use Route Reflectors instead of a node-to-node mesh,
+you'll need to SSH into a UCP node and download the `calicoctl` command.
+
+Log in to a UCP node using SSH, and run:
+
+```
+sudo curl --location https://github.com/projectcalico/calicoctl/releases/download/v3.1.1/calicoctl \
+  --output /usr/bin/calicoctl
+sudo chmod +x /usr/bin/calicoctl
+```
+
+Now you need to configure `calicoctl` to communicate with the etcd key-value
+store managed by UCP. Create a file named `/etc/calico/calicoctl.cfg` with
+the following content:
+
+```
+apiVersion: projectcalico.org/v3
+kind: CalicoAPIConfig
+metadata:
+spec:
+  datastoreType: "etcdv3"
+  etcdEndpoints: "127.0.0.1:12379"
```

Comment:
this is my fault. The port is 12378

Reply:
I was testing this and `12379` seems to be the right one.

```
docker ps --format '{{.Names}} | {{.Image}} | {{.Ports}}' | grep ucp-kv

ucp-kv | docker/ucp-etcd:3.0.0 | 2380/tcp, 4001/tcp, 7001/tcp, 0.0.0.0:12380->12380/tcp, 0.0.0.0:12379->2379/tcp
```

Reply:
But we should point to the proxy port in case of non managers. Or we make sure we mention that this is to be executed on the master node only.